engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/TemplateDataFactoryImpl.java

```diff
@@ -68,7 +68,11 @@
 
     @Override
     public TemplateInfo getTemplateOnPrimaryStorage(long templateId, DataStore store, String configuration) {
-        VMTemplateVO templ = imageDataDao.findById(templateId);
+        VMTemplateVO templ = imageDataDao.findByIdIncludingRemoved(templateId);
```

Comment:
@nvazquez `getTemplateOnPrimaryStorage` is also used by `PrimaryDataStoreImpl::getTemplate`. Will using `findByIdIncludingRemoved` here would affect those workflows?

Reply:
@shwstppr before PR #4748 a template would never be marked as removed in DB when was destroyed, but since this PR is updating the removed field, the getTemplate() method wouldn't be able to access the template for the storage cleanup tasks. Which other use case are you considering?

Reply:
@nvazquez thanks for answering. My doubt was regarding `createManagedStorageVolumeFromTemplateAsync` from VolumeServiceImpl which tries to find the template on primary storage. But it seems that won't get affected. Setting this as resolved.